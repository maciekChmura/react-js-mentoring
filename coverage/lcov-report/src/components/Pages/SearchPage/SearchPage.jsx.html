<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Pages/SearchPage/SearchPage.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/components/Pages/SearchPage</a> SearchPage.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">21.43% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.18% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">30% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/10</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { SearchCSSGrid, HeaderCSSGrid } from './SearchPage.Styles';
import { fetchFromSearch, fetchDefault } from '../../../utils/dataLoaders';
import {
  sortingTypeForSearch,
  sortingTypeForDisplay
} from '../../../utils/sortingTypeConstants';
import PageName from '../../Header/PageName/PageName';
import FormTitle from '../../Header/FormTitle/FormTitle';
import SearchForm from '../../Header/SearchForm/SearchForm';
import Results from '../../Body/Results/Results';
import ResultsOptions from '../../Helper/ResultsOptions/ResultsOptions';
&nbsp;
class SearchPage extends Component {
  state = {
    data: '',
    sortingType: 'release date',
    searchOption: 'title'
  };
&nbsp;
  componentDidMount = () =&gt; {
    fetchDefault(12).then(<span class="fstat-no" title="function not covered" >da</span>ta =&gt;
<span class="cstat-no" title="statement not covered" >      this.setState(<span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{ data: data.data })</span>)</span>
    );
  };
&nbsp;
  changeSorting = <span class="fstat-no" title="function not covered" >da</span>ta =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState(<span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{ sortingType: data })</span>);</span>
  };
&nbsp;
  performSearch = <span class="fstat-no" title="function not covered" >se</span>archString =&gt; {
    const { sortingType, searchOption } = <span class="cstat-no" title="statement not covered" >this.state;</span>
<span class="cstat-no" title="statement not covered" >    fetchFromSearch(</span>
      searchString,
      sortingTypeForSearch[sortingType],
      searchOption
    ).then(<span class="fstat-no" title="function not covered" >da</span>ta =&gt; <span class="cstat-no" title="statement not covered" >this.setState(<span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{ data: data.data })</span>))</span>;
  };
&nbsp;
  changeSearch = <span class="fstat-no" title="function not covered" >da</span>ta =&gt; {
<span class="cstat-no" title="statement not covered" >    event.preventDefault();</span>
<span class="cstat-no" title="statement not covered" >    this.setState(<span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{ searchOption: data })</span>);</span>
  };
&nbsp;
  render() {
    const { data, sortingType, searchOption } = this.state;
&nbsp;
    return (
      &lt;SearchCSSGrid&gt;
        &lt;HeaderCSSGrid&gt;
          &lt;PageName /&gt;
          &lt;FormTitle /&gt;
          &lt;SearchForm
            handleFormSubmit={this.performSearch}
            searchOption={searchOption}
            changeSearch={this.changeSearch}
          /&gt;
        &lt;/HeaderCSSGrid&gt;
        &lt;ResultsOptions
          dataSize={data.length}
          changeSorting={this.changeSorting}
          sortingType={sortingTypeForDisplay[sortingType]}
        /&gt;
        {data ? <span class="branch-0 cbranch-no" title="branch not covered" >&lt;Results results={data} /&gt; </span>: &lt;p&gt;loading&lt;/p&gt;}
      &lt;/SearchCSSGrid&gt;
    );
  }
}
&nbsp;
export default SearchPage;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Dec 21 2018 15:25:26 GMT+0100 (STD)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
