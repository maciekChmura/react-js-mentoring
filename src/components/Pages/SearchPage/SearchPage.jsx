import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getSearchData } from '../../../redux/actions';
import {
  sortingTypeForSearch,
  sortingTypeForDisplay
} from '../../../utils/sortingTypeConstants';
import PageName from '../../Header/PageName/PageName';
import FormTitle from '../../Header/FormTitle/FormTitle';
import SearchForm from '../../Header/SearchForm/SearchForm';
import Results from '../../Body/Results/Results';
import ResultsOptions from '../../Helper/ResultsOptions/ResultsOptions';
import LoadingWrapper from '../../Helper/Loading/Loading';

import { SearchCSSGrid, HeaderCSSGrid } from './SearchPage.Styles';

const mapStateToProps = state => ({
  movies: state.movies,
  sortingType: state.sortingType,
  searchOption: state.searchOption,
  isSearching: state.isSearching,
  error: state.isSearching,
  searchValue: state.searchValue
});

export class SearchPage extends Component {
  // static getDerivedStateFromProps(props, state) {
  //   const { term } = props.match.params; // receiving current search term from URL
  //   const { searchValue } = props; // receiving stored search term
  //   const { sortingType, searchOption, getSearchData, error } = props;
  //   console.log(term);
  //   if (term !== searchValue) {
  //     // checking whether we should perform new search or not
  //     getSearchData(term, sortingTypeForSearch[sortingType], searchOption);
  //   }

  //   return state;
  // }

  performSearch = searchString => {
    const { term } = this.props.match.params;
    const { sortingType, searchOption, getSearchData, error } = this.props;
    console.log(term);

    getSearchData(
      searchString,
      sortingTypeForSearch[sortingType],
      searchOption
    );
    if (error) {
      console.log('Search failed');
    }
  };

  render() {
    const { movies, sortingType, isSearching } = this.props;
    return (
      <SearchCSSGrid>
        <HeaderCSSGrid>
          <PageName />
          <FormTitle />
          <SearchForm handleFormSubmit={this.performSearch} />
        </HeaderCSSGrid>
        <ResultsOptions sortingType={sortingTypeForDisplay[sortingType]} />
        {isSearching ? (
          <LoadingWrapper />
        ) : (
          movies.length !== 0 && <Results results={movies} />
        )}
      </SearchCSSGrid>
    );
  }
}

export default connect(
  mapStateToProps,
  { getSearchData }
)(SearchPage);
