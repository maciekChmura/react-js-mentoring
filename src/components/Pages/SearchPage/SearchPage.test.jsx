import React from 'react';
import { shallow, mount } from 'enzyme';

import SearchPage from './SearchPage';

describe('SearchPage', () => {
  it('should render correctly', () => {
    const component = shallow(<SearchPage />);

    expect(component).toMatchSnapshot();
  });

  it('should render loading spinner when no data is in state', () => {
    const component = shallow(<SearchPage />);
    expect(component.find('Results')).toHaveLength(0);
  });

  it('should render results when data for movies is in state', () => {
    const component = shallow(<SearchPage />);
    expect(component.find('Results')).toHaveLength(0);
    component.setState({ data: 'movies data' });
    expect(component.find('Results')).toHaveLength(1);
  });

  it('changeSorting changes state when called', () => {
    const component = shallow(<SearchPage />);

    component.find('ResultsOptions').prop('changeSorting')('sortingOne');
    expect(component.state('sortingType')).toEqual('sortingOne');
  });

  it('changeSearch changes state when called', () => {
    const component = shallow(<SearchPage />);

    component.find('SearchForm').prop('changeSearch')('searchingOne');
    expect(component.state('searchOption')).toEqual('searchingOne');
  });

  // it('should set state on mount', () => {
  //   const component = mount(<SearchPage />);

  //   expect(component.setState('state')).toBeCalled();
  // });
});
