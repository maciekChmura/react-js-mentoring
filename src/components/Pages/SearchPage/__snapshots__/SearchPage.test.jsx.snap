// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <PageName />
          <FormTitle
            title="FIND YOUR MOVIE"
          />
          <SearchForm
            changeSearch={[Function]}
            handleFormSubmit={[Function]}
            searchOption="title"
          />
        </ForwardRef>,
        <ResultsOptions
          changeSorting={[Function]}
          dataSize={0}
          sortingType="release date"
        />,
        <p>
          loading
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <PageName />,
            <FormTitle
              title="FIND YOUR MOVIE"
            />,
            <SearchForm
              changeSearch={[Function]}
              handleFormSubmit={[Function]}
              searchOption="title"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "FIND YOUR MOVIE",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "changeSearch": [Function],
              "handleFormSubmit": [Function],
              "searchOption": "title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "SearchPageStyles__HeaderCSSGrid-sc-1ueu3tk-1",
            "isStatic": true,
            "rules": Array [
              "grid-area:2 / 3 / 1 / 2;display:grid;grid-template-columns:50px 800px 50px;grid-template-rows:80px 50px 100px;background-image:url(",
              "test-file-stub",
              ");background-position:center;background-repeat:no-repeat;background-size:cover;",
            ],
          },
          "displayName": "SearchPageStyles__HeaderCSSGrid",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "SearchPageStyles__HeaderCSSGrid-sc-1ueu3tk-1",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "changeSorting": [Function],
          "dataSize": 0,
          "sortingType": "release date",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "loading",
        },
        "ref": null,
        "rendered": "loading",
        "type": "p",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "SearchPageStyles__SearchCSSGrid-sc-1ueu3tk-0",
        "isStatic": true,
        "rules": Array [
          "display:grid;grid-template-columns:1fr 900px 1fr;grid-template-rows:230px 50px auto;",
        ],
      },
      "displayName": "SearchPageStyles__SearchCSSGrid",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "SearchPageStyles__SearchCSSGrid-sc-1ueu3tk-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <PageName />
            <FormTitle
              title="FIND YOUR MOVIE"
            />
            <SearchForm
              changeSearch={[Function]}
              handleFormSubmit={[Function]}
              searchOption="title"
            />
          </ForwardRef>,
          <ResultsOptions
            changeSorting={[Function]}
            dataSize={0}
            sortingType="release date"
          />,
          <p>
            loading
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <PageName />,
              <FormTitle
                title="FIND YOUR MOVIE"
              />,
              <SearchForm
                changeSearch={[Function]}
                handleFormSubmit={[Function]}
                searchOption="title"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "title": "FIND YOUR MOVIE",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "changeSearch": [Function],
                "handleFormSubmit": [Function],
                "searchOption": "title",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "SearchPageStyles__HeaderCSSGrid-sc-1ueu3tk-1",
              "isStatic": true,
              "rules": Array [
                "grid-area:2 / 3 / 1 / 2;display:grid;grid-template-columns:50px 800px 50px;grid-template-rows:80px 50px 100px;background-image:url(",
                "test-file-stub",
                ");background-position:center;background-repeat:no-repeat;background-size:cover;",
              ],
            },
            "displayName": "SearchPageStyles__HeaderCSSGrid",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "SearchPageStyles__HeaderCSSGrid-sc-1ueu3tk-1",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeSorting": [Function],
            "dataSize": 0,
            "sortingType": "release date",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "loading",
          },
          "ref": null,
          "rendered": "loading",
          "type": "p",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "SearchPageStyles__SearchCSSGrid-sc-1ueu3tk-0",
          "isStatic": true,
          "rules": Array [
            "display:grid;grid-template-columns:1fr 900px 1fr;grid-template-rows:230px 50px auto;",
          ],
        },
        "displayName": "SearchPageStyles__SearchCSSGrid",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "SearchPageStyles__SearchCSSGrid-sc-1ueu3tk-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
